### PR TITLE
refactor: import jest as global; unify import style of some modules

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

I spent a while playing with running Jest in ESM. I was very close to success. Now, I'm a bit more confident about how to roll this out gradually—as usual, the migration (even when it's only for development) will be split into several granular PRs so people can follow the progress and catch any bugs in the process.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Tests still pass

## Related PRs

#6520

### CODE REVIEW

File: packages/docusaurus-theme-common/src/utils/announcementBarUtils.tsx [hunk 1]
```diff
@@ -96,7 +95,9 @@ const useAnnouncementBarContextValue = (): AnnouncementBarAPI => {
   );
 };
 
-const AnnouncementBarContext = createContext<AnnouncementBarAPI | null>(null);
+const AnnouncementBarContext = React.createContext<AnnouncementBarAPI | null>(
```
Comment (slorber) [inline packages/docusaurus-theme-common/src/utils/announcementBarUtils.tsx:98]: not against this change, just wondering why it's necessary?

File: .eslintrc.js [hunk 2]
```diff
@@ -312,5 +317,11 @@ module.exports = {
         '@typescript-eslint/explicit-module-boundary-types': OFF,
       },
     },
+    {
+      files: ['*.test.ts', '*.test.tsx'],
+      rules: {
+        'import/no-extraneous-dependencies': OFF,
```
Comment (slorber) [inline .eslintrc.js:323]: maybe this should be disabled with more granularity (if possible)? 🤷‍♂️

File: packages/docusaurus-theme-common/src/utils/announcementBarUtils.tsx (hunk 1 above)
Comment (Josh-Cena) [reply packages/docusaurus-theme-common/src/utils/announcementBarUtils.tsx:98]: Oh, purely stylistic. We have some `React.createContext` and some `createContext`. I personally prefer to have the `React.` prefix except for hooks so we understand what we are talking about

File: .eslintrc.js (hunk 2 above)
Comment (Josh-Cena) [reply .eslintrc.js:323]: Test files are internal anyways. If they can run, there's nothing to worry about.

**Pull request merged**